### DISCUSSION TITLE
defineStore actions 'this' is typed 'any'

### DISCUSSION THREAD

**Discussion by nurdism** [vuejs/pinia#965] [Help and Questions] [answered]
I'm using this in a nuxt3 project and I'm running `this.x` being typed as `any`, is this normal? do I have to create a `this` type and assignee it to the action (ie `reset(this: ApplicationStore)`). I have pinia@2.0.9 installed.

https://i.imgur.com/PmDrHv5.png

```ts
import { defineStore } from 'pinia'

export const useApplicationStore = defineStore({
  id: 'applicationStore',
  state: () => ({
    id: '',
  }),
  actions: {
    reset() {
      this.id = ''
    },
  },
})
```

**Comment by tre-dev** [comment]
Same problem. Any solution?

**Comment by nurdism** [comment, marked as answer]
I ended opening up an issue about it https://github.com/vuejs/pinia/issues/981 `{ strict: true }` or try `{ noImplicitThis: true }` in your TS config.
